### DISCUSSION TITLE
What should be in the Readme?

### DISCUSSION THREAD

**Discussion by lionztt** [apache/incubator-hugegraph#2229] [General]
我这边有几个问题想在社区中寻求讨论和帮助。
I have a few issues that I would like to discuss and help in the community.

 什么内容需要放在 Readme 里面，可以让大家更清晰了解hugegraph？
What content needs to be placed in Readme, so that everyone can understand Hugegraph more clearly?

多长篇幅的 Readme 适宜阅读，能够关注到足够的重点又不失去耐心呢？
 How long is Readme suitable for reading, can you pay attention to focus on the right points without losing patience?

目前我这边提了一个pull request，详情请看[pull request 连接](https://github.com/apache/incubator-hugegraph/pull/2227)，更改了部分 Readme 内容，欢迎大家一起讨论。
Now I have put forward a pull request, please see the details [here](https://github.com/apache/incubator-hugegraph/pull/2227), some Readme content has been changed, welcome to discuss together.

**Comment by imbajin** [comment]
we could refer other graduated/top ASF projects first:
- [dubbo](https://github.com/apache/dubbo)
- [spark](https://github.com/apache/spark)
- [echarts](https://github.com/apache/echarts)
- [superset](https://github.com/apache/superset)
- [kafka](https://github.com/apache/kafka)
- [pulsar](https://github.com/apache/pulsar)
- ...

then refer other incubating projects, and everyone could give some suggestion here 💬
 @javeme @simon824 @JackyYangPassion @wanganjuan @zyxxoo @liuxiaocs7

**Comment by liuxiaocs7** [comment]
Hi, community, I've browsed the READMEs of the above projects and some other Apache projects, the common contents include brief introduction, features, build process, contribution guide, License, community contact points. Some may also include usage examples, and keep the README content as concise as possible, unnecessary unnecessary content can be put into the official website instead of the README.
For the changes in PR #2227, I think the diagrams of Contributor and Stargazers, Contributors graph are unnecessary (maybe in website is better?) and License should not be removed.
@imbajin What do you think?

**Reply by imbajin** [reply]
team (include PMC + committers + contributors) could show like [this](https://streampark.apache.org/team)

<img width="408" alt="image" src="https://github.com/apache/incubator-hugegraph/assets/17706099/3f301ed4-5c8a-4816-8a06-a008ae0d4308">

**Reply by lionztt** [reply]
I removed the separate license because there is a small icon at the beginning of the ReadMe to indicate the license of the project. Maybe this icon is not obvious enough, I will add the text version of the license back. 

I think Contributor and Stargazers graph maybe not unnecessary,and that 2 charts is removed now. But I think we need to retain contributors graph in the ReadMe ,which can encourage and acknowledge our contributors, if you think it takes too much space, I think we can put a slightly smaller size of the image.

The contributors graph is the following picture.
[![contributors graph](https://contrib.rocks/image?repo=apache/hugegraph)](https://github.com/apache/incubator-hugegraph/graphs/contributors)

**Reply by liuxiaocs7** [reply]
Thanks for the detailed explanation, sorry i overlooked the icon, it looks fine to me now 😄

**Comment by imbajin** [comment]
And also we do need an **architecture picture** in the README & website
